src/types.rs

```diff
@@ -128,3 +138,77 @@ impl ResponseParam {
         }
     }
 }
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct EventRequest {
+    pub tx_hash: DisplayBytes,
+    pub network: String,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct TxLog {
+    pub address: DisplayBytes,
+    pub data: DisplayBytes,
+    pub index: String,
+    pub topics: [Option<DisplayBytes>; 4],
```

Comment:
Why not to use `Vec<Hash>` as is done in `ethabi::log::RawLog` you are using in `process_events`?